### PR TITLE
Do not overwrite multiple FROMs for multi-stage builds

### PR DESCRIPTION
**Pull request by lcarva**
Since 17.05 multiple FROMs are possible see
https://docs.docker.com/engine/reference/builder/#from

### CODE REVIEW

File: dockerfile_parse/parser.py [hunk 1]
```diff
@@ -494,6 +494,8 @@ def _modify_instruction(self, instruction, new_value):
                 del lines[insn['startline']:insn['endline'] + 1]
                 lines.insert(insn['startline'], new_line)
                 self.lines = lines
+                if insn['instruction'] == 'FROM':  # Only overwrite first base-image
```
Comment (twaugh) [inline dockerfile_parse/parser.py]: Could be `if instruction == 'FROM':`, and that might make the intent clearer too.

File: tests/test_parser.py [hunk 2]
```diff
@@ -236,6 +236,27 @@ def test_modify_instruction(self, dfparser):
...
+CMD  {1}
+
+FROM {2}
+CMD  {3}""").format(FROM[0], CMD[0], FROM[1], CMD[1])
+
+        dfparser.content = df_content
+
+        assert dfparser.baseimage == FROM[0]
+        dfparser.baseimage = FROM[1]
+        assert dfparser.baseimage == FROM[1]
```
[the remark this answers is not in the document]
Comment (lcarva) [reply tests/test_parser.py]: Great catch! Fixed.

File: dockerfile_parse/parser.py (hunk 1 above)
Comment (lcarva) [reply dockerfile_parse/parser.py]: Fixed.

**Comment by lcarva** [comment]
@twaugh, @rcerven: comments have been addressed. I also did some additional enhancements to `test_parser.py`. Let me know what you think (on Monday, that is 😉 ).

**Comment by twaugh** [comment]
Holding off merging until the next release.

**Comment by lcarva** [comment]
Updated release notes draft:
> Do not overwrite multiple FROMs for multi-stage builds

**Pull request merged**